src/metabase/api/search.clj

```diff
@@ -182,21 +185,24 @@
   (str "%" s "%"))
 
 (defn- search-string-clause
-  [query searchable-columns]
+  [model query]
   (when query
-    (into [:or]
-          (for [column searchable-columns
-                token (scoring/tokenize (scoring/normalize query))]
-            [:like
-             (hsql/call :lower column)
-             (wildcard-match token)]))))
+    (let [tokens (scoring/tokenize (scoring/normalize query))]
+      (into [:or]
+            (apply concat (for [column (search-config/searchable-columns-for-model model)
+                                :let [qualified-column (hsql/qualify (model->alias model) column)]]
+                            (if (and (= column :dataset_query)
+                                     (= (mdb/db-type) :postgres))
+                              [(hsql/raw ["dataset_query_tokens @@ to_tsquery(" #sql/param :search-tokens ")"])]
```

Comment:
Note this #sql/param syntax is required to allow parameters in raw sql, see https://github.com/seancorfield/honeysql/commit/61d8926a63e3ca7c9c1abf55b76d883e2cd2244d#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R278

Reply:
same note about `plainto_tsquery`

also, kinda looks like a sql injection? can we do that raw call by concat thing?